horace_core/sqw/axes_blocks/@AxesBlockBase/private/bin_pixels_.m

```diff
@@ -215,22 +215,10 @@
 end
 pix = pix_ok;
 
-if ndims > 0
-    if nout == 6
-        if ~isa(pix.data,'double') && force_double % TODO: this should be moved to get_pixels
-            pix = PixelDataBase.create(double(pix.data));     % when PixelData is separated into file accessor and memory accessor
-        end
-    else % sort will make pix double if requested  TODO: this should be moved to get_pixels
-        pix = sort_pix(pix,pix_indx,npix1,varargin{:}); % when PixelData is separated into file accessor and memory accessor
-    end
-elseif ndims == 0
-    if ~isa(pix.data,'double') && force_double % TODO: this should be moved to get_pixels
-        pix = PixelDataBase.create(double(pix.data));     % when PixelData is separated into file accessor and memory accessor
-    end
-
-    if nout == 6
-        pix_indx = ones(pix.num_pixels,1);
-    end
+% TODO: this should be moved to get_pixels
+% when PixelData is separated into file accessor and memory accessor
+if ndims > 0 && nout ~= 6
+    pix = sort_pix(pix, pix_indx, npix1, varargin{:});
```

Comment:
I'm not 100% sure and this was the logic of the original, but I believe that because when `nout==6` we're asking for the indices of those points added in order to sort them later. Sorting them now would be a waste of time.